include/range/v3/view/cartesian_product.hpp

```diff
@@ -237,6 +237,17 @@ namespace ranges
                     return check_at_end_(meta::size_t<N - 1>{}, at_end ||
                         std::get<N - 1>(its_) == ranges::end(std::get<N - 1>(view_->views_)));
                 }
+                cursor(end_tag, constify_if<cartesian_product_view> &view, std::true_type) // Bounded
+                  : cursor(begin_tag{}, view)
+                {
+                    std::get<0>(its_) = ranges::end(std::get<0>(view.views_));
+                }
+                cursor(end_tag, constify_if<cartesian_product_view> &view, std::false_type) // !Bounded
+                  : cursor(begin_tag{}, view)
+                {
```

Comment:
`// Only called when the 0th view type is !Bounded && RandomAccess && Sized` would save readers a few minutes determining that this is O(1).